Modules/_cursesmodule.c

```diff
@@ -4140,18 +4159,16 @@ static PyObject *
 _curses_resize_term_impl(PyObject *module, int nlines, int ncols)
 /*[clinic end generated code: output=9e26d8b9ea311ed2 input=2197edd05b049ed4]*/
 {
-    PyObject *result;
-
     PyCursesInitialised;
 
-    result = PyCursesCheckERR(resize_term(nlines, ncols), "resize_term");
-    if (!result)
-        return NULL;
-    if (!update_lines_cols()) {
-        Py_DECREF(result);
-        return NULL;
-    }
+    PyObject *result = PyCursesCheckERR(resize_term(nlines, ncols), "resize_term");
+    CHECK_NOT_NULL_OR_ERROR(result);
+    CHECK_RET_FLAG_OR_ERROR(update_lines_cols(module));
```

Comment:
I don't understand this macro. I would prefer:

`if (!update_lines_cols(module)) goto error;`

`CHECK_RET_FLAG_OR_ERROR` + goto makes the code less straightforward.

Reply:
Maybe my naming sense is terrible but it was something I found in the code sometimes (e.g., `CHECK_INT` in `_decimal.c`). But I'll change the macros since it's probably better to reduce the diff rather than reducing the overall number of lines.

Reply:
EDIT: I think know why the macro was confusing. I'll remove it and use an explicit `if (!(...))`.